examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+
+    Optional<Attribute<MediaPlayback.CurrentState>> currentStateAttribute = mediaPlayback.getCurrentState();
+    if(currentStateAttribute.isPresent() && currentStateAttribute.get().isAvailable())
+    {
+        CompletableFuture<MediaPlayback.CurrentState> currentStateValue = currentStateAttribute.get().read();
+        try {
+            Log.i(TAG, "Read Current State value: " + currentStateValue.get(1, TimeUnit.SECONDS));
+        } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            Log.e(TAG, "Exception when reading CurrentState " + e);
+        }
+    }
+    else
+    {
+        Log.e(TAG, "Attribute unavailable on the selected endpoint");
+    }
+}
+else
+{
+    Log.e(TAG, "Required cluster not found on the selected endpoint");
+}
+```
+
+On iOS, an attribute can be read with implementation like the one below.
+
+```objectivec
+if(endpoint.hasCluster(MTREndpointClusterTypeContentLauncher))
+{
+    let cluster: MTRMediaPlaybackCluster = endpoint.cluster(for: MTREndpointClusterTypeMediaPlayback) as! MTRMediaPlaybackCluster
+    let currentStateAttribute: MTRAttribute<MTRCurrentState> = cluster.currentState
+
+    currentStateAttribute.read { currentStateValue, err in
+        if(err == nil)
+        {
+            self.status = "Read CurrentState value \(String(describing: currentStateValue))"
+        }
+        else
+        {
+            self.status = "Error when reading CurrentState value \(String(describing: err))"
+        }
+    }
+}
+else
+{
+    self.status = "No MediaPlayback cluster supporting endpoint found"
+}
+```
+
+### Subscriptions
+
+_{Complete Attribute subscription examples:
+[Android](android/App/app/src/main/java/com/matter/casting/AttributeSubscriptionExample.java)
+|
+[iOS](darwin/TvCasting/TvCasting/MTRAttributeSubscriptionExampleViewModel.swift)}_
+
+A Casting Client may subscribe to an attribute from the `contentAppEndpoint` on
+the CastingPlayer to get event notifications related to that attribute.
+
+On Android, an attribute can be subscribed to in the following way.
+
+```java
+// check if the Endpoint supports the Cluster to use
+if (contentAppEndpoint.get().hasCluster(MediaPlayback.class)) {
+    // get the Cluster to use
+     MediaPlayback mediaPlayback = contentAppEndpoint.get().getCluster(MediaPlayback.class);
+
+    Optional<Attribute<MediaPlayback.CurrentState>> currentStateAttribute = mediaPlayback.getCurrentState();
+    if(currentStateAttribute.isPresent() && currentStateAttribute.get().isAvailable())
+    {
+        currentStateAttribute.get().addObserver(new Attribute.Listener() {
+            @Override
+            public void onError(Error error) {
+                Log.e(TAG, "Error when listening to CurrentState " + error);
+            }
+
+            @Override
+            public void onChange(Object currentStateValue) {
+                Log.i(TAG, "CurrentState changed to value " + currentStateValue);
+            }
+        }, 0, 1);
```

Comment:
Can you replace 0 and 1 here by a readable constant?
```suggestion
}, MIN_INTERVAL, MAX_INTERVAL);
```

Reply:
Done